### PR TITLE
Add common empty state component

### LINKED ISSUE #1129: Implement new empty state
New unified empty state was designed and need update on all screens where we use them. Everything is specified here: https://github.com/Joystream/atlas/issues/866

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #1129 
https://www.figma.com/file/TGUpBXgXf9ceMvHbRB5BIx/Joystream-Atlas?node-id=12179%3A381509

### CODE REVIEW

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import styled from '@emotion/styled'
+
+import { Text } from '@/shared/components'
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: Components inside `shared` should use relative imports
```suggestion
import { Text } from '../Text'
```

File: src/shared/components/EmptyFallback/EmptyFallback.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    height: 152,
+  },
+}
+
+export const EmptyFallback: FC<EmptyFallbackProps> = ({ title, subtitle, variant = 'large', button }) => (
+  <Container variant={variant}>
+    <SvgEmptyStateIllustration width={ILLUSTRATION_SIZES[variant].width} height={ILLUSTRATION_SIZES[variant].height} />
+    <Message>
+      {title && <Title variant="h4">{title}</Title>}
+      {subtitle && <Subtitle single={!title}>{subtitle}</Subtitle>}
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.tsx]: Why do we need the `single` prop? The design seems to handle this the following way:
a) variant is `large`, use `h4` + secondary `body2`
b) variant is `small`, use primary `body1`

Following this policy, you also don't need to provide any custom font styles, just use the `Text` component.

File: src/shared/components/EmptyFallback/EmptyFallback.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import { SvgEmptyStateIllustration } from '@/shared/illustrations'
+
+import { Container, Message, Subtitle, Title } from './EmptyFallback.styles'
+
+export type EmptyFallbackSizes = 'small' | 'large'
+
+export type EmptyFallbackProps = {
+  title?: string
+  subtitle?: string | null
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.tsx]: Let's make `title` required and use it in the `small` variant

File: src/views/studio/MyUploadsView/MyUploadsView.tsx [hunk 4]
```diff
@@ -49,7 +51,15 @@ export const MyUploadsView: React.FC = () => {
       ) : hasUploads ? (
         groupedUploadsState.map((files) => <UploadStatusGroup key={files[0].parentObject.id} uploads={files} />)
       ) : (
-        <EmptyUploadsView />
+        <EmptyFallback
+          title="No ongoing uploads"
+          subtitle="You will see statuses of ongoing uploads for each asset you transfer."
```
Comment (kdembler) [inline src/views/studio/MyUploadsView/MyUploadsView.tsx]:
```suggestion
          subtitle="You will see status of each ongoing upload here"
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 5]
```diff
@@ -254,7 +254,15 @@ export const MyVideosView = () => {
       <ViewContainer>
         <Text variant="h2">My videos</Text>
         {hasNoVideos ? (
-          <EmptyVideosView />
+          <EmptyFallback
+            title="Add your first video"
+            subtitle="Your channel is empty! Start the publisher journey by adding your very first video to Joystream."
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
            subtitle="No videos uploaded yet. Start publishing by adding your first video to Joystream."
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 6]
```diff
@@ -285,15 +293,34 @@ export const MyVideosView = () => {
             </Grid>
             {((isDraftTab && drafts.length === 0) ||
               (!isDraftTab && !loading && totalCount === 0 && (!videos || videos.length === 0))) && (
-              <EmptyVideos
-                text={
+              <EmptyFallback
+                title={
+                  currentTabName === 'All Videos'
+                    ? 'No videos here yet'
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
                    ? 'No videos yet'
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 7]
```diff
@@ -285,15 +293,34 @@ export const MyVideosView = () => {
...
+                    : currentTabName === 'Public'
+                    ? 'No public videos here yet'
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
                    ? 'No public videos yet'
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 8]
```diff
@@ -285,15 +293,34 @@ export const MyVideosView = () => {
...
                   currentTabName === 'All Videos'
-                    ? "You don't have any published videos at the moment"
+                    ? null
                     : currentTabName === 'Public'
-                    ? "You don't have any public videos at the moment"
+                    ? 'Upload your first public video'
                     : currentTabName === 'Drafts'
-                    ? "You don't have any drafts at the moment"
-                    : "You don't have any unlisted videos at the moment"
+                    ? 'Each unfinished project will be saved here as a draft. Start publishing to see something here.'
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
                    ? 'Each video that hasn't been published yet will be available here as a draft.'
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 9]
```diff
@@ -285,15 +293,34 @@ export const MyVideosView = () => {
...
+                    : 'To make one choose “unlisted” option in privacy settings of the video, or use video tile menu.'
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
                    : 'Videos published with "Unlisted" privacy setting will show up here.'
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 10]
```diff
@@ -285,15 +293,34 @@ export const MyVideosView = () => {
...
+                    ? 'No drafts here yet'
+                    : 'No unlisted videos here yet'
+                }
+                subtitle={
                   currentTabName === 'All Videos'
-                    ? "You don't have any published videos at the moment"
+                    ? null
                     : currentTabName === 'Public'
-                    ? "You don't have any public videos at the moment"
+                    ? 'Upload your first public video'
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
                    ? 'Videos published with "Public" privacy setting will show up here.'
```

File: src/views/viewer/ChannelView/ChannelView.tsx [hunk 11]
```diff
@@ -214,6 +215,12 @@ export const ChannelView: React.FC = () => {
         return (
           <>
             <VideoSection className={transitions.names.slide}>
+              {!videosWithPlaceholders.length && isSearching && (
+                <EmptyFallback subtitle={`No videos on the channels matching query “${searchQuery}”`} variant="small" />
+              )}
+              {!videosWithPlaceholders.length && !isSearching && (
+                <EmptyFallback subtitle="No videos found on the channel" variant="small" />
```
Comment (kdembler) [inline src/views/viewer/ChannelView/ChannelView.tsx]:
```suggestion
                <EmptyFallback subtitle="No videos on this channel" variant="small" />
```

File: src/views/viewer/ChannelView/ChannelView.tsx [hunk 12]
```diff
@@ -214,6 +215,12 @@ export const ChannelView: React.FC = () => {
         return (
           <>
             <VideoSection className={transitions.names.slide}>
+              {!videosWithPlaceholders.length && isSearching && (
+                <EmptyFallback subtitle={`No videos on the channels matching query “${searchQuery}”`} variant="small" />
```
Comment (kdembler) [inline src/views/viewer/ChannelView/ChannelView.tsx]:
```suggestion
                <EmptyFallback subtitle={`No videos matching "${searchQuery}" query found`} variant="small" />
```

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+import { EmptyFallbackSizes } from './EmptyFallback'
+
+export const Container = styled.div<{ variant?: EmptyFallbackSizes }>`
+  margin: ${({ variant }) => (variant === 'large' ? sizes(20) : sizes(13.5))} auto 0;
+  display: grid;
+  place-items: center;
+
+  ${media.compact} {
+    width: ${sizes(90)};
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: This should likely be set only for the `large` variant

**Review by kdembler — CHANGES_REQUESTED** [review]

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 14]
```diff
@@ -0,0 +1,38 @@
+import styled from '@emotion/styled'
+
+import { colors, media, sizes } from '@/shared/theme'
+
+import { EmptyFallbackSizes } from './EmptyFallback'
+
+import { Text } from '../Text'
+
+export const Container = styled.div<{ variant?: EmptyFallbackSizes }>`
+  margin: ${({ variant }) => (variant === 'large' ? sizes(20) : sizes(13.5))} auto 0;
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: Where do these margins come from? The design seems to have a constant 40px vertical margin

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+  > svg {
+    max-width: 650px;
+  }
+`
+
+export const Message = styled.div`
+  display: flex;
+  flex-direction: column;
+  text-align: center;
+  margin-bottom: ${sizes(6)};
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: If we apply 40px margin on `Container` I don't think we need this

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+`
+
+export const Title = styled(Text)`
+  line-height: 1.25;
+  margin-top: ${sizes(10)};
+`
+export const Subtitle = styled(Text)`
+  margin-top: ${sizes(2)};
+  line-height: ${sizes(5)};
+  color: ${colors.gray[300]};
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: This isn't needed, we should use `secondary` prop on `Text`

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+export const Message = styled.div`
+  display: flex;
+  flex-direction: column;
+  text-align: center;
+  margin-bottom: ${sizes(6)};
+`
+
+export const Title = styled(Text)`
+  line-height: 1.25;
+  margin-top: ${sizes(10)};
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: This changes based on the variant. I suggest to apply this to `Message` instead

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+  display: grid;
+  place-items: center;
+
+  ${media.compact} {
+    width: ${({ variant }) => (variant === 'large' ? sizes(90) : 'auto')};
+  }
+
+  > svg {
+    max-width: 650px;
+  }
```
Comment (kdembler) [inline src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: Is this needed? We apply explicit sizes to the SVG anyway right?

**Review by kdembler — CHANGES_REQUESTED** [review]

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts (hunk 15 above)
Comment (rafalpawlow) [reply src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: Actually this is the space between message and button, margin on container will not handle this.

Comment (kdembler) [reply src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: Let's apply that margin to the button then, `Message` will not always have button after it, right

File: src/shared/components/EmptyFallback/EmptyFallback.styles.ts (hunk 14 above)
Comment (rafalpawlow) [reply src/shared/components/EmptyFallback/EmptyFallback.styles.ts]: That's what I suggested, but ok I will change this
Zrzut ekranu 2021-08-11 o 10 40 28

**Review by kdembler — APPROVED** [review]

**Pull request merged**